Sources/Regex/RegexParse.swift

```diff
@@ -120,12 +122,46 @@ extension Parser {
       let child = try parse()
       partialResult = isCapturing ? .capturingGroup(child) : .group(child)
       try lexer.eat(expecting: .rightParen)
-    case .character(let c, _)?:
+
+    case .character(let c, isEscaped: false):
       lexer.eat()
       partialResult = .character(c)
+
+    case .character("u", isEscaped: true):
+      lexer.eat()
+      var digits = ""
+      for _ in 0..<4 {
```

Comment:
Any reason not to have the lexer tokenize this for us?

Reply:
The lexer should definitely do this - I'll move it up and add a couple other formats.